keps/sig-node/5075-dra-dynamic-provisioning-support/README.md

```diff
@@ -0,0 +1,1055 @@
+- Enhance a capability of secondary networks to dynamically allocate secondary networks based on present availabilities such as bandwidth.
+- Leverage the enhancement of capacity consumption from [KEP-4815](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4815-dra-partitionable-devices).
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+- Define device-specific or vendor-specific attributes.
+- Support network security policy.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+Users can request for secondary networks based on their bandwidth demands without the need to concern about neither device name nor a present available bandwidth of each specific device. DRA scheduler can dynamically select available device based on its availability. 
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaim
+...
+spec:
+  devices:
+    provisions:
+    - name: net1
+      deviceClassName: vlan-cni.networking.x-k8s.io
+      resources:
+        bandwidth:
+          quantity: 10Gi
+```
+
+#### Story 2
+Users can make sure that fractions of memory are reserved for the virtual GPU.
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaim
+...
+spec:
+  devices:
+    provisions:
+    - name: vgpu0
+      deviceClassName: vgpu.nvidia.com
+      resources:
+        memory:
+          quantity: 10Gi
+```
+
+#### Story 3
+Users can specify a device source.
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaim
+...
+spec:
+  devices:
+    provisions:
+    - name: net1
+      deviceClassName: vlan-cni.networking.x-k8s.io
+      selectors:
+      - cel:
+          expression: "device.attributes['vlan-cni.networking.x-k8s.io'].name == 'eth1'"
+      resources:
+        bandwidth:
+          quantity: 1Gi
+```
+#### Story 4
+Users can define more than one provision requests to get devices from different sources. From the following claim spec, the claim should provide two devices which have different device sources.
+
+```yaml
+apiVersion: resource.k8s.io/v1beta1
+kind: ResourceClaim
+...
+spec:
+  devices:
+    provisions:
+    - name: net1
+      deviceClassName: vlan-cni.networking.x-k8s.io
```

Comment:
I think you'll also need to attach some config to specify the VLAN ID, for example

Reply:
Yes, the CNI configuration will be also required for ResourceClaim as @LionelJouin's design in cni-dra-driver. 
https://github.com/kubernetes-sigs/cni-dra-driver/blob/eb94de8c61835de1f13eefe700de9d9d89615531/docs/design/device-configuration-api.md#design

I will add reference in this KEP too.